### PR TITLE
Refactor IntegrityStmt

### PR DESCRIPTION
**Pull request by tohrnii**
This PR refactors the `IntegrityStmt` to make it easy to incorporate selectors.

### CODE REVIEW

File: ir/src/constraint_builder/integrity_constraints/mod.rs [hunk 1]
```diff
@@ -4,9 +4,10 @@ use super::{
     ListFoldingValueType, NamedTraceAccess, SemanticError, Symbol, SymbolType, Variable,
     VariableType, VectorAccess, CURRENT_ROW,
 };
+// TODO: Move this to air-script-core
+use parser::ast::ConstraintType;
```
Comment (grjte) [inline ir/src/constraint_builder/integrity_constraints/mod.rs:8]: Does it make sense to just put this in `air-script-core` now?

File: parser/src/parser/tests/trace_columns.rs [hunk 2]
```diff
@@ -1,3 +1,5 @@
+use crate::ast::ConstraintType;
+
```
Comment (grjte) [inline parser/src/parser/tests/trace_columns.rs:2]: nit: Let's remove the extra line here

File: ir/src/constraint_builder/integrity_constraints/mod.rs (hunk 1 above)
Comment (tohrnii) [reply ir/src/constraint_builder/integrity_constraints/mod.rs:8]: Sorry, I didn't realize then that we will have to move all it's dependencies to `air-script-core` (which would essentially be almost everything in parser). So, I have just imported from `super::ast` instead. Please let me know if you disagree with the change.

**Comment by tohrnii** [comment]
Sorry, I merged the PRs in wrong order. The selector parser PR was merged into this branch instead.

**Review by grjte — APPROVED** [review]
Looks good, thanks!

File: ir/src/constraint_builder/integrity_constraints/mod.rs (hunk 1 above)
Comment (grjte) [reply ir/src/constraint_builder/integrity_constraints/mod.rs]: Makes sense!

**Pull request merged**